File: Intro/ts/fizzBuzz.ts
export function fizzBuzz(n: number): Array<number | string> {
  const result: Array<number | string> = [];
  for (let i = 1; i <= n; i++) {
    if (i % 15 === 0) {
      result.push("fizzbuzz");
    } else if (i % 3 === 0) {
      result.push("fizz");
    } else if (i % 5 === 0) {
      result.push("buzz");
    } else {
      result.push(i);
    }
  }
  return result;
}

File: Intro/csharp/FizzBuzz.cs
using System.Collections.Generic;

namespace IntroSolutions;

public static class FizzBuzz
{
    public static IList<object> Solve(int n)
    {
        var result = new List<object>();
        for (int i = 1; i <= n; i++)
        {
            bool fizz = i % 3 == 0;
            bool buzz = i % 5 == 0;
            if (fizz && buzz)
            {
                result.Add("fizzbuzz");
            }
            else if (fizz)
            {
                result.Add("fizz");
            }
            else if (buzz)
            {
                result.Add("buzz");
            }
            else
            {
                result.Add(i);
            }
        }

        return result;
    }
}
